libs/blocks/preflight/checks/asoApi.js

```diff
@@ -0,0 +1,277 @@
+import { SEO_TITLES, STATUS, SEO_IDS, SEO_DESCRIPTIONS } from './constants.js';
+
+const CHECK_API = 'https://spacecat.experiencecloud.live/api/v1';
+
+export const asoCache = {
+  identify: null,
+  suggest: null,
+  suggestPromise: null,
+  tokenPromise: null,
+};
+
+const lanaLog = (message) => {
+  window.lana.log(message, {
+    sampleRate: 1,
+    tags: 'preflight',
+    errorType: 'e',
+  });
+};
+
+async function getASOToken() {
+  asoCache.tokenPromise = asoCache.tokenPromise || (async () => {
```

Comment:
Sir Nit-a-lot says: maybe a bit less repetition on this?
```suggestion
  asoCache.tokenPromise ??= (async () => {
```

Reply:
Nice nit 👍